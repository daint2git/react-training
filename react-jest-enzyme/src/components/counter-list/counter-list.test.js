import React from 'react'
import { shallow, mount, render } from 'enzyme'
import renderer from 'react-test-renderer'
import CounterList from './counter-list'

describe('CounterList', () => {
  test('matches the snapshot', () => {
    const tree = renderer.create(<CounterList />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('render correctly (shallow)', () => {
    const tree = shallow(<CounterList />)
    expect(tree).toMatchSnapshot()
  })

  test('render correctly (mount)', () => {
    const tree = mount(<CounterList />)
    expect(tree).toMatchSnapshot()
  })

  test('render correctly (render)', () => {
    const tree = render(<CounterList />)
    expect(tree).toMatchSnapshot()
  })

  // test('debug shallow vs mount', () => {
  //   const shallowWrapper = shallow(<CounterList />)
  //   const mountWrapper = mount(<CounterList />)

  //   console.log('shallow html', shallowWrapper.debug())
  //   console.log('mount html', mountWrapper.debug())
  // })

  test('should render two counters by default', () => {
    const wrapper = shallow(<CounterList />)
    const counters = wrapper.find('Counter')
    expect(counters.length).toEqual(2)
  })

  test('add more counters when user click the button', () => {
    const wrapper = shallow(<CounterList />)
    const addBtn = wrapper.find('Button')
    addBtn.simulate('click')
    const counters = wrapper.find('Counter')
    expect(counters.length).toEqual(3)
  })

  test('initialNumberCounters props is 10', () => {
    const wrapper = shallow(<CounterList initialNumberCounters={10} />)
    expect(wrapper.instance().props).toEqual({ initialNumberCounters: 10 })
  })

  test('initialNumberCounters props is 10 (2)', () => {
    const wrapper = shallow(<CounterList initialNumberCounters={10} />)
    expect(wrapper.instance().props.initialNumberCounters).toEqual(10)
  })

  test('numberCounters state is 10 when initialNumberCounters props is 10', () => {
    const wrapper = shallow(<CounterList initialNumberCounters={10} />)
    expect(wrapper.state().numberCounters).toEqual(10)
  })
})
