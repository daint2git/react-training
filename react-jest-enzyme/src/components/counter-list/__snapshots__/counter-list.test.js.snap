// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterList matches the snapshot 1`] = `
Array [
  <ul>
    <li>
      <div
        className="counter"
      >
        <p>
          Current count: 
          0
        </p>
        <button
          name="increment"
          onClick={[Function]}
        >
          increment
        </button>
        <button
          name="decrement"
          onClick={[Function]}
        >
          decrement
        </button>
      </div>
    </li>
    <li>
      <div
        className="counter"
      >
        <p>
          Current count: 
          0
        </p>
        <button
          name="increment"
          onClick={[Function]}
        >
          increment
        </button>
        <button
          name="decrement"
          onClick={[Function]}
        >
          decrement
        </button>
      </div>
    </li>
  </ul>,
  <button
    className="sc-bdVaJa ihDGdR"
    onClick={[Function]}
    type="button"
  >
    add Counter
  </button>,
]
`;

exports[`CounterList render correctly (mount) 1`] = `
Array [
  <ul>
    <li
      key="0"
    >
      <div
        className="counter"
      >
        <p>
          Current count: 
          0
        </p>
        <button
          name="increment"
          onClick={[Function]}
        >
          increment
        </button>
        <button
          name="decrement"
          onClick={[Function]}
        >
          decrement
        </button>
      </div>
    </li>
    <li
      key="1"
    >
      <div
        className="counter"
      >
        <p>
          Current count: 
          0
        </p>
        <button
          name="increment"
          onClick={[Function]}
        >
          increment
        </button>
        <button
          name="decrement"
          onClick={[Function]}
        >
          decrement
        </button>
      </div>
    </li>
  </ul>,
  <styled.button
    onClick={[Function]}
    type="button"
  >
    <button
      className="sc-bdVaJa ihDGdR"
      onClick={[Function]}
      type="button"
    >
      add Counter
    </button>
  </styled.button>,
]
`;

exports[`CounterList render correctly (render) 1`] = `
Array [
  <ul>
    <li>
      <div
        class="counter"
      >
        <p>
          Current count: 0
        </p>
        <button
          name="increment"
        >
          increment
        </button>
        <button
          name="decrement"
        >
          decrement
        </button>
      </div>
    </li>
    <li>
      <div
        class="counter"
      >
        <p>
          Current count: 0
        </p>
        <button
          name="increment"
        >
          increment
        </button>
        <button
          name="decrement"
        >
          decrement
        </button>
      </div>
    </li>
  </ul>,
  <button
    class="sc-bdVaJa ihDGdR"
    type="button"
  >
    add Counter
  </button>,
]
`;

exports[`CounterList render correctly (shallow) 1`] = `
<Fragment>
  <ul>
    <li
      key="0"
    >
      <Counter />
    </li>
    <li
      key="1"
    >
      <Counter />
    </li>
  </ul>
  <Button
    onClick={[Function]}
  >
    add Counter
  </Button>
</Fragment>
`;
